src/Appwrite/Migration/Version/V20.php

```diff
@@ -45,19 +44,228 @@ public function execute(): void
         Console::log('Migrating Project: ' . $this->project->getAttribute('name') . ' (' . $this->project->getId() . ')');
         $this->projectDB->setNamespace("_{$this->project->getInternalId()}");
 
+        Console::info('Migrating Collections');
+        $this->migrateCollections();
+
         Console::info('Migrating Functions');
         $this->migrateFunctions();
 
         Console::info('Migrating Databases');
         $this->migrateDatabases();
 
-        Console::info('Migrating Collections');
-        $this->migrateCollections();
-
         Console::info('Migrating Buckets');
         $this->migrateBuckets();
+
+        Console::info('Migrating Documents');
+        $this->forEachDocument([$this, 'fixDocument']);
+    }
+
+    /**
+     * Migrate Collections.
+     *
+     * @return void
+     * @throws Exception|Throwable
+     */
+    private function migrateCollections(): void
+    {
+        $internalProjectId = $this->project->getInternalId();
+        $collectionType = match ($internalProjectId) {
+            'console' => 'console',
+            default => 'projects',
+        };
+
+        // Support database array type migration (user collections)
+        $attributes = $this->projectDB->find('attributes', [
+            Query::equal('array', [true]),
+        ]);
+
+        foreach ($attributes as $attribute) {
+            $indexes = $this->projectDB->find('indexes', [
+                Query::equal('databaseInternalId', [$attribute['databaseInternalId']]),
+                Query::equal('collectionInternalId', [$attribute['collectionInternalId']]),
+            ]);
+            $foundIndex = 0;
+            foreach ($indexes as $index) {
+                if (in_array($attribute['key'], $index['attributes'])) {
+                    $this->projectDB->deleteIndex($index['collectionId'], $index['_uid']);
+                    $foundIndex = 1;
+                }
+            }
+            if ($foundIndex === 1) {
+                $this->projectDB->updateAttribute($attribute['collectionInternalId'], $attribute['key'], Database::VAR_STRING);
```

Comment:
Let's use $attribute['type'] instead of hard-coding string to catch integer/float/other attribute types too